### PR TITLE
Adding ability to use react router Links in nav items

### PR DESCRIPTION
**Pull request by AFed29**
Worked on alongside @mcgoddard, @simonireilly and @MirceaUngureanu

Trying to resolve issue #49

### CODE REVIEW

**Comment by AFed29** [comment]
Would love to get your thoughts about this @Tomdango

File: src/components/header/components/NavItem.tsx [hunk 1]
```diff
@@ -3,22 +3,32 @@ import classNames from 'classnames';
 import { ChevronRight as ChevronRightIcon } from '../../icons';
 
 export interface NavItemProps extends HTMLProps<HTMLAnchorElement> {
+  asElement?: React.ElementType;
+  to?: string;
```
Comment (Tomdango) [inline src/components/header/components/NavItem.tsx:7]: Could we potentially move the `asElement` and `to` declarations into a shared interface somewhere else in the codebase (i.e. maybe a new file in `util/types`)? I was thinking something like:
```ts
interface AsElementLink<T extends HTMLElement> extends HTMLProps<T> {
    asElement?: React.ElementType;
    to?: string;
}
```

**Review by Tomdango — CHANGES_REQUESTED** [review]
This looks good - it appears to use the same pattern that a lot of other popular libraries use so it's a tried and tested pattern.

File: src/components/header/components/NavItem.tsx (hunk 1 above)
Comment (AFed29) [reply src/components/header/components/NavItem.tsx]: Refactored to pull out the interface into a separate file

**Review by Tomdango — APPROVED** [review]
LGTM :+1: I'll get an updated package out today with these changes in

**Comment by charlesomer** [comment]
Just a note, I don't think the storybook online has been updated to reflect these changes? Please correct me if I'm wrong though!

**Comment by charlesomer** [comment]
Also, does this work for the `Button` component too? Or does that require `as` instead of `asElement`?

**Comment by AFed29** [comment]
> Just a note, I don't think the storybook online has been updated to reflect these changes? Please correct me if I'm wrong though!

We've only updated the Header stories, to use a custom component however this doesn't seem to be on the online Storybook @Tomdango ? We deliberately chose not to include react-router links just to make sure that we didn't couple that dependancy to these components at all. We only put in a generic example where we have passed a simple div. If any other examples are needed for other components then could we raise another issue? 


> Also, does this work for the `Button` component too? Or does that require `as` instead of `asElement`?

We didn't touch the button component, I'm a bit wary of changing its functionality. Maybe something for another issue too?

**Pull request merged**